contents/tutorials/redirect-testing.md

```diff
@@ -0,0 +1,383 @@
+2. Create a user ID and bootstrap it to the client if not.
+
+This requires using your project API key to get the cookies, parsing them as JSON, and potentially creating a distinct ID using `crypto.randomUUID()`. Altogether, this looks like this:
+
+```js
+import { NextResponse } from 'next/server'
+
+export async function middleware(request) {
+  const ph_project_api_key = '<ph_project_api_key>'
+  const ph_cookie_key = `ph_${ph_project_api_key}_posthog`
+  const cookie = request.cookies.get(ph_cookie_key);
+
+  let distinct_id;
+  if (cookie) {
+    distinct_id = JSON.parse(cookie.value).distinct_id;
+  } else {
+    distinct_id= crypto.randomUUID();
+  }
+//... rest of code
+```
+
+### Evaluating our redirect test with PostHog
+
+With our distinct ID, we can use the PostHog API to evaluate the `main-redirect` feature flag we set up earlier (because we can’t use PostHog SDKs in Next.js middleware). 
+
+We evaluate the flag by making a POST request to the `https://app.posthog.com/decide?v=3` route (replace `app` with `eu` if you’re on EU Cloud) with your project API key and user distinct ID. From the response, we get the value of the `main-redirect` feature flag and use it to redirect to the right page. Altogether, this looks like this:
+
+```js
+//... rest of code
+
+	const requestOptions = {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify({
+      api_key: ph_project_api_key,
+      distinct_id: distinct_id
+    })
+  };
+  
+  const ph_request = await fetch(
+    'https://app.posthog.com/decide?v=3', // or eu
+    requestOptions
+  );
+  const data = await ph_request.json();
+
+  const flagResponse = data.featureFlags['main-redirect']
+
+  if (request.nextUrl.pathname=== '/test' && flagResponse === 'control') {
+    return NextResponse.redirect(new URL('/control', request.url))
+  }
+  if (request.nextUrl.pathname=== '/control' && flagResponse === 'test') {
+    return NextResponse.redirect(new URL('/test', request.url))
+  }
+  return NextResponse.next()
+}
+//... rest of code
+```
+
+### Capturing exposure
+
+To get accurate results for our experiment, we also need to capture a `$feature_flag_called` exposure event. This requires another POST request. We can add this in like this:
+
+```js
+//... rest of code
+const flagResponse = data.featureFlags['main-redirect']
+
+// Capture events, for exposure
+const eventOptions = {
+  method: 'POST',
+  headers: {
+    'Content-Type': 'application/json'
+  },
+  body: JSON.stringify({
+    api_key: ph_project_api_key,
+    distinct_id: distinct_id,
+    properties: {
+      "$feature_flag": 'main-redirect',
+      "$feature_flag_response": flagResponse
+    },
+    event: "$feature_flag_called"
+  })
+};
+
+const eventRequest = await fetch(
+  'https://app.posthog.com/capture',
+  eventOptions
+);
+
+if (request.nextUrl.pathname === '/test' && flagResponse === 'control') {
+//... rest of code
+```
+
+## Bootstrapping the data
```

Comment:
Do we need this section? I know it's a cool feature we have, but I feel its a bit of an advanced technique and not strictly necessary for creating redirect tests. My concern is that it makes the tutorial harder to follow

Reply:
We do, I added a callout explainer on why.